File: SESION3/dart/filtro-ubicacion.dart
import 'package:sesion_3_ds/filtro.dart';
import 'package:sesion_3_ds/alquiler.dart';

class FiltroUbicacion extends Filtro {
  FiltroUbicacion(){}

  void aplicarFiltro(Alquiler alq) {
    String loc = alq.getLocalizacion();
    
    for (final x in ubicaciones){
      if (x.getLocalizacion() == loc) 
        alq.incrementarCoste(x.getPrecioBase());
    }
    
  }

File: SESION3/C++/FiltroUbicacion.cpp
using namespace std;

#include "FiltroUbicacion.h"
#include "Ubicacion.h"
#include "Alquiler.h"
#include "Filtro.h"



void FiltroUbicacion::aplicarFiltro(Alquiler &alq) {

    string loc= alq.getLocalizacion();

    for( auto item : ubicaciones){
        if(item.getLocalizacion() == loc){
            alq.incrementarCoste(item.getPrecioBase());
        }
    }

}


